components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -131,6 +136,80 @@ void BravePrefProvider::ShutdownOnUIThread() {
   PrefProvider::ShutdownOnUIThread();
 }
 
+// static
+void BravePrefProvider::RegisterProfilePrefs(
+    user_prefs::PrefRegistrySyncable* registry) {
+  PrefProvider::RegisterProfilePrefs(registry);
+  // Register shields settings migration pref.
+  registry->RegisterIntegerPref(kBraveShieldsSettingsVersion, 1);
+}
+
+void BravePrefProvider::MigrateShieldsSettings(bool incognito) {
+  // Incognito inherits from regular profile, so nothing to do.
+  if (incognito)
+    return;
+  MigrateShieldsSettingsV1ToV2();
+}
+
+void BravePrefProvider::MigrateShieldsSettingsV1ToV2() {
+  // Check if migration is needed.
+  if (prefs_->GetInteger(kBraveShieldsSettingsVersion) != 1)
+    return;
+
+  // All sources in ContentSettingsType::PLUGINS we want to migrate.
+  for (const auto& resource_id : GetShieldsResourceIDs()) {
+    MigrateShieldsSettingsV1ToV2ForOneType(ContentSettingsType::PLUGINS,
+                                           resource_id);
+  }
+
+  // ContentSettingsType::JAVASCRIPT.
+  MigrateShieldsSettingsV1ToV2ForOneType(ContentSettingsType::JAVASCRIPT,
+                                         std::string());
+
+  // Mark migration as done.
+  prefs_->SetInteger(kBraveShieldsSettingsVersion, 2);
+}
+
+void BravePrefProvider::MigrateShieldsSettingsV1ToV2ForOneType(
+    ContentSettingsType content_type,
+    const std::string& resource_id) {
+  using OldRule = std::pair<ContentSettingsPattern, ContentSettingsPattern>;
+  // Find rules that can be migrated and create replacement rules for them.
+  std::vector<OldRule> old_rules;
+  std::vector<Rule> new_rules;
+  auto rule_iterator = PrefProvider::GetRuleIterator(content_type, resource_id,
+                                                     /*off_the_record*/ false);
+  while (rule_iterator && rule_iterator->HasNext()) {
+    auto rule = rule_iterator->Next();
+    auto new_primary_pattern =
+        ConvertPatternToWildcardSchemeAndPort(rule.primary_pattern);
+    auto new_secondary_pattern =
+        ConvertPatternToWildcardSchemeAndPort(rule.secondary_pattern);
+    if (new_primary_pattern || new_secondary_pattern) {
+      old_rules.emplace_back(rule.primary_pattern, rule.secondary_pattern);
+      new_rules.emplace_back(
+          new_primary_pattern.value_or(rule.primary_pattern),
+          new_secondary_pattern.value_or(rule.secondary_pattern),
+          rule.value.Clone());
+    }
+  }
+  rule_iterator.reset();
+
+  // Migrate.
+  DCHECK(old_rules.size() == new_rules.size());
```

Comment:
DCHECK_EQ

Reply:
Fixed in bdc2851b5386b1ae42a134550a217edb6267ef57